File: Locator/src/Locator.Contracts/Vacancies/Dto/FullVacancyDto.cs
namespace Locator.Contracts.Vacancies.Dto;

public record FullVacancyDto
{
    public FullVacancyDto(long id, VacancyDto dto, double? rating)
    {
        Id = id;
        Name = dto.Name;
        Url = dto.Url;
        Responsibility = dto.Description?.Responsibility;
        Employer = dto.Employer?.Name;
        Area = dto.Area;
        Address = dto.Address;
        Experience = dto.Experience;
        Salary = dto.Salary;
        Schedule = dto.Schedule;
        WorkFormat = dto.WorkFormat;
        Rating = rating;
    }

    public long Id { get; init; }
    public string Name { get; init; }
    public string Url { get; init; }
    public string? Responsibility { get; init; }
    public string? Employer { get; init; }
    public Area? Area { get; init; }
    public Address? Address { get; init; }
    public Experience? Experience { get; init; }
    public Salary? Salary { get; init; }
    public Schedule? Schedule { get; init; }
    public List<WorkFormat>? WorkFormat { get; init; }
    public double? Rating { get; init; }
}

File: Locator/client/src/contracts/dtos/FullVacancyDto.ts
export interface Area {
    id: string;
    name: string;
    url?: string;
}

export interface Address {
    city?: string;
    street?: string;
    building?: string;
    description?: string;
    lat?: number;
    lng?: number;
    raw?: string;
}

export interface Experience {
    id: string;
    name: string;
}

export interface Salary {
    from?: number;
    to?: number;
    currency: string;
    gross?: boolean;
}

export interface Schedule {
    id: string;
    name: string;
}

export interface WorkFormat {
    id: string;
    name: string;
}

export interface FullVacancyDto {
    id: number;
    name: string;
    url: string;
    responsibility?: string;
    employer?: string;
    area?: Area;
    address?: Address;
    experience?: Experience;
    salary?: Salary;
    schedule?: Schedule;
    workFormat?: WorkFormat[];
    rating?: number;
}